src/test/java/core/basesyntax/service/impl/FileReaderCsvTest.java

```diff
@@ -0,0 +1,95 @@
+
+    @Test
+    void read_validPath_Ok() {
+        List<String> actual = fileReaderCsv.read(DEFAULT_VALID_FILE);
+        Assertions.assertEquals(actual, fileReaderCsv.read(DEFAULT_VALID_FILE));
+    }
+
+    @Test
+    void read_invalidPath_notOk() {
+        Throwable exception = Assertions.assertThrows(InvalidDataException.class, () -> {
+            fileReaderCsv.read(INVALID_PATH_TO_FILE);
+        });
+        Assertions.assertEquals("Can't read file " + INVALID_PATH_TO_FILE, exception.getMessage());
+    }
+
+    @Test
+    void parse_validData_Ok() {
+        List<FruitTransaction> actual = fruitTransactionParser.parse(fileReaderCsv
+                .read(DEFAULT_VALID_FILE));
+        List<FruitTransaction> expected = new ArrayList<>();
+        Assertions.assertSame(expected.getClass(), actual.getClass());
+    }
+
+    @Test
+    void parse_invalidQuantity_notOk() {
+        Throwable exception = Assertions.assertThrows(InvalidDataException.class, () -> {
+            fruitTransactionParser.parse(fileReaderCsv.read(INVALID_QUANTITY_IN_FILE));
+        });
+        Assertions.assertEquals("Input quantity is not valid", exception.getMessage());
+    }
+
+    @Test
+    void parse_NullQuantity_notOk() {
+        Throwable exception = Assertions.assertThrows(InvalidDataException.class, () -> {
+            fruitTransactionParser.parse(fileReaderCsv.read(NULL_QUANTITY_IN_FILE));
+        });
+        Assertions.assertEquals("Input quantity is not valid", exception.getMessage());
+    }
+
+    @Test
+    void parse_NullFruit_notOk() {
+        Throwable exception = Assertions.assertThrows(InvalidDataException.class, () -> {
+            fruitTransactionParser.parse(fileReaderCsv.read(NULL_FRUIT_IN_FILE));
+        });
+        Assertions.assertEquals("Fruit does not exist", exception.getMessage());
+    }
+
+    @Test
+    void generateReport_validData_Ok() {
+        String actual = reportGenerator.generateReport(Storage.fruits);
+        String expected = "expected";
+        Assertions.assertSame(expected.getClass(), actual.getClass());
+    }
+
+    @Test
+    void execute_validData_Ok() {
+        Map<FruitTransaction.Operation, OperationHandler> strategyMap = new HashMap<>();
+        strategyMap.put(FruitTransaction.Operation.BALANCE, new BalanceHandler());
+        strategyMap.put(FruitTransaction.Operation.PURCHASE, new PurchaseHandler());
+        strategyMap.put(FruitTransaction.Operation.RETURN, new ReturnHandler());
+        strategyMap.put(FruitTransaction.Operation.SUPPLY, new SupplyHandler());
+        TransactionExecutor transactionExecutor = new TransactionExecutor(strategyMap);
+        transactionExecutor.execute(fruitTransactionParser.parse(fileReaderCsv
+                .read(DEFAULT_VALID_FILE)));
+        Assertions.assertEquals(transactionExecutor.getStrategy()
+                .get(FruitTransaction.Operation.BALANCE).getClass(), BalanceHandler.class);
+    }
+}
```

Comment:
In my opinion, it's not advisable to use methods from your application directly in your test methods. Instead, it's better to utilize basic API methods for testing purposes.